demos/stardist_demo/python/stardist_demo.py

```diff
@@ -0,0 +1,373 @@
+        self.height = meta.get_SizeY()
+
+        # Find an index of channel
+        self.channel_id = None
+        for i in range(meta.channel_count):
+            if meta.Channel(i).Name == channel:
+                self.channel_id = i
+                break
+        if self.channel_id is None:
+            raise Exception('Unable to find channel ' + channel)
+
+        self.reader = bf.ImageReader(path)
+
+
+    def get_tile(self, tile_id, sz):
+        x, y = tile_id
+        y = min(y * sz, self.height - sz)  # Last tile offset
+        x = min(x * sz, self.width - sz)
+        data = self.reader.read(XYWH=((x, y, sz, sz)), rescale=False)
+        return data[:, :, self.channel_id:self.channel_id + 1]
+
+
+    def get_preview(self, max_width):
+        return None, None
+
+
+class Contour:
+    def __init__(self, pts):
+        self.pts = pts
+        self.xmin, self.ymin = np.amin(pts, axis=0)
+        self.xmax, self.ymax = np.amax(pts, axis=0)
+        mask = np.zeros((self.ymax - self.ymin + 1,
+                         self.xmax - self.xmin + 1), dtype=np.uint8)
+        self.mask = cv.drawContours(mask, [pts - [self.xmin, self.ymin]], -1, (1), cv.FILLED)
+
+
+def normalize_percentile(inp, percentiles=[1.0, 99.0]):
+    num_channels = inp.shape[2]
+    total = inp.shape[0] * inp.shape[1]
+    num_colors = 256
+    scale = []
+    offset = []
+    for ch in range(num_channels):
+        # Compute a histogram for a channel
+        hist_item = cv.calcHist([inp], [ch], None, [num_colors], [0, num_colors])
+
+        # Find two percentiles from colors distribution
+        counter = 0
+        i = 0
+        ind = int(percentiles[i] / 100.0 * total)
+        rng = [0, 0]
+        for color in range(num_colors):
+            counter += hist_item[color]
+            if counter >= ind:
+                rng[i] = color
+                if i == 1:
+                    break
+                else:
+                    i += 1
+                ind = int(percentiles[i] / 100.0 * total)
+        scale.append(1.0 / (rng[1] - rng[0]))
+        offset.append(-rng[0])
+
+    return (inp + offset) * scale
+
+
+def nms(coords, probs, nms_threshold, tile_size):
+    # To optimize NMS we compare contours not with each other but with a global
+    # predictions mask
+    global_mask = np.zeros((tile_size, tile_size), dtype=np.uint8)
+
+    coords = [np.clip(pts, 0, tile_size - 1) for pts in coords]
+    contours = [Contour(pts) for pts in coords]
+    ids = []
+    for i in reversed(np.argsort(probs)):
+        ctr = contours[i]
+        ref_mask = global_mask[ctr.ymin : ctr.ymax + 1, ctr.xmin : ctr.xmax + 1]
+
+        inter_area = np.count_nonzero(np.logical_and(ref_mask, ctr.mask))
+        union_area = np.count_nonzero(np.logical_or(ref_mask, ctr.mask))
+        iou = inter_area / union_area
+        if iou <= nms_threshold:
+            ids.append(i)
+            ref_mask[:, :] |= ctr.mask[:, :]
+
+    return ids
+
+
+class StarDistModel:
```

Comment:
Please remove `Model` suffix and move the class and corresponding functions like `normalize_percentile()` into common with other models.